Services/UI/OJS.Services.Ui.Models/Contests/ContestForListingServiceModel.cs

```diff
@@ -29,9 +30,16 @@ public class ContestForListingServiceModel : IMapExplicitly
 
     public double OrderBy { get; set; }
 
+    public TimeSpan? Duration { get; set; }
+
+    public int NumberOfProblems { get; set; }
+
     public void RegisterMappings(IProfileExpression configuration)
         => configuration.CreateMap<Contest, ContestForListingServiceModel>()
             .ForMember(
                 dest => dest.Category,
-                opt => opt.MapFrom(src => src.Category!.Name));
+                opt => opt.MapFrom(src => src.Category!.Name))
+            .ForMember(
+                dest => dest.NumberOfProblems,
+                opt => opt.MapFrom(src => src.ProblemGroups.First().Problems.Count));
```

Comment:
This does not seem correct. If you have 4 ProblemGroups with 3 Problems in each group, how many problems will the user have in the contest?

Reply:
Yes, it was not correct. I now calculate the number of problems taking the count of problem groups, where each group must have at least one problem in it.